pallets/loans/src/lib.rs

```diff
@@ -580,25 +639,32 @@ impl<T: Config> Pallet<T> {
         Ok(Ratio::from_rational(borrows, total))
     }
 
-    pub fn update_borrow_index(
-        borrow_rate_per_block: Rate,
-        currency_id: CurrencyId,
-    ) -> DispatchResult {
-        // borrowIndex = borrowIndex * borrowRate + borrowIndex
+    pub fn update_borrow_index(borrow_rate: Rate, currency_id: CurrencyId) -> DispatchResult {
+        // interestAccumulated = totalBorrows * borrowRate
+        // totalBorrows = interestAccumulated + totalBorrows
+        // totalReserves = interestAccumulated * reserveFactor + totalReserves
+        // borrowIndex = borrowIndex * (1 + borrowRate)
         let borrows_prior = Self::total_borrows(currency_id);
-        let interest_accumulated = borrow_rate_per_block
+        let reserve_prior = Self::total_reserves(currency_id);
+        let reserve_factor = Self::reserve_factor(currency_id);
+        let interest_accumulated = borrow_rate
             .checked_mul_int(borrows_prior)
             .ok_or(Error::<T>::CalcAccrueInterestFailed)?;
         let total_borrows_new = interest_accumulated
             .checked_add(borrows_prior)
             .ok_or(Error::<T>::CalcAccrueInterestFailed)?;
+        let total_reserves_new = reserve_factor
+            .mul_floor(interest_accumulated)
+            .checked_add(reserve_prior)
+            .ok_or(Error::<T>::CalcAccrueInterestFailed)?;
         let borrow_index = Self::borrow_index(currency_id);
         let borrow_index_new = borrow_index
-            .checked_mul(&borrow_rate_per_block)
+            .checked_mul(&borrow_rate)
             .and_then(|r| r.checked_add(&borrow_index))
             .ok_or(Error::<T>::CalcAccrueInterestFailed)?;
 
         TotalBorrows::<T>::insert(currency_id, total_borrows_new);
+        TotalReserves::<T>::insert(currency_id, total_reserves_new);
```

Comment:
when will the money goes into our reserve account?

Reply:
It's a topic of governance, and we can discuss how to use the reserve.